### DISCUSSION TITLE
Issues with running Container Resources on Linux

### DISCUSSION THREAD

**Discussion by AlexanderBartoshZ** [microsoft/aspire#5573] [Q&A] [answered]
Hey all! 
Aspire is a really nice effort!

High level use case is very simple: 
Run aspire test on Github actions workers (or any other CI-CD linux based worker). 
Aspire Workload version is 8.2

My specific case would be GitHub actions Acr controller running DinD runners. 
My simplified Aspire Host application is: 
`builder.AddKafka("kafka").WithKafkaUI();`

1. AFAIU _Server: Docker Engine - Community_ is not supported Container resource runner, and indeed you get all the complains from kafka UI not being able to resolve host.docker.internal .. to connect to kafka
2. When trying podman KakfaUI still is unable to connect to Kafka.  
` Connection to node -1 (host.containers.internal/10.88.0.1:44633) could not be established. Broker may not be available.`

When zooming in on why I see that dcpctrl is listening on 127.0.0.1 instead of 0.0.0.0  
`tcp        0      0 127.0.0.1:44633         0.0.0.0:*               LISTEN      53329/dcpctrl `
packages/aspire.hosting.orchestration.linux-x64/8.2.0/tools/ext/dcpctrl

My weird conclusion is that apparently a container can not connect to any other proxied service when running on Linux? 
That is a bit too weird to be true!

Any help is greatly appreciated! 

Thank you, 
Alexander

**Comment by AlexanderBartoshZ** [comment]
Related : 
https://github.com/dotnet/aspire/issues/5584
https://github.com/dotnet/aspire/issues/4136

@karolz-ms is there any way to instruct the dcpctrl to listen on all interfaces instead of 127.0.0.1 for proxying? 
https://github.com/dotnet/aspire/issues/4136#issuecomment-2329474139

Thank you !

**Reply by karolz-ms** [reply, marked as answer]
Alexander, sorry for late reply.

We are working to make it possible to do so (i.e. bind to all interfaces via `0.0.0.0` pseudo-address, or bind to specific network interface) in Aspire 9 release (ETA ~couple of months from now). In the currently-available Aspire releases it is kind of buggy

**Reply by AlexanderBartoshZ** [reply]
Thank you! 
Had to avoid Proxying to make it work
